packages/core/src/modal/modal.stories.ts

```diff
@@ -23,118 +24,265 @@ export default {
 };
 
 export function API(args: any) {
+  let initted = false;
+  const modalId = 'api-modal';
+
+  function showApiModal() {
+    const myModal = document.getElementById(modalId) as CdsModal;
+    myModal.removeAttribute('hidden');
+
+    if (!initted) {
+      myModal.addEventListener('closeChange', () => {
+        myModal.setAttribute('hidden', 'true');
+      });
+      initted = true;
+    }
+  }
+
   return html`
     <cds-demo popover>
-      <cds-modal __demo-mode ...="${spreadProps(getElementStorybookArgs(args))}">
+      <cds-button status="primary" type="button" @click=${showApiModal}>Show Modal</cds-button>
+      <cds-modal ...="${spreadProps(getElementStorybookArgs(args))}" id="${modalId}">
         <cds-modal-header>
           <h3 cds-text="title">${args['cds-modal-header']}</h3>
         </cds-modal-header>
         <cds-modal-content>
           <p cds-text="body">${args['cds-modal-content']}</p>
         </cds-modal-content>
-        <cds-modal-actions>
-          ${args['cds-modal-actions']}
-        </cds-modal-actions>
+        <cds-modal-actions> ${args['cds-modal-actions']} </cds-modal-actions>
       </cds-modal>
     </cds-demo>
   `;
 }
 
-/** @website */
 export function defaultSize() {
```

Comment:
Would the interactive demos be better as new additional stories? They are good and helpful for focus trap testing. I think the main demos should remain stateless since it helps show the component differences and the code snippets for the docs. The code is marked for the website docs so we likely would want them as plain html when possible. Will the `__demo-mode` continue to work for the stateless demos? It will be needed for story screenshot testing and maybe the website as well?